extension/scripts/background.js

```diff
@@ -67,21 +67,28 @@ async function convertDatabase() {
 
 		for (const key in defaultStorage) {
 			if (!oldStorage) oldStorage = {};
-			if (!key in oldStorage) oldStorage[key] = {};
+			if (!(key in oldStorage)) oldStorage[key] = {};
 
 			if (typeof defaultStorage[key] === "object") {
 				if (defaultStorage[key] instanceof DefaultSetting) {
 					let useCurrent = true;
 
-					if (defaultStorage[key].type === "array") {
-						if (!Array.isArray(oldStorage[key])) {
-							useDefault();
-							useCurrent = false;
-						}
-					} else if (
-						!defaultStorage[key].type.split("|").some((value) => value === typeof oldStorage[key] || (value === "empty" && oldStorage[key] === ""))
+					if (

```

Comment:
I don't think I like this either. `useDefault` is so much easier to understand what it actually does. It makes the condition overcomplicated imo as well.